src/heap_utils.cc

```diff
@@ -231,19 +234,178 @@ class BufferOutputStream : public v8::OutputStream {
   std::unique_ptr<JSString> buffer_;
 };
 
+namespace {
+class FileOutputStream : public v8::OutputStream {
+ public:
+  explicit FileOutputStream(FILE* stream) : stream_(stream) {}
+
+  virtual int GetChunkSize() {
+    return 65536;  // big chunks == faster
+  }
+
+  virtual void EndOfStream() {}
+
+  virtual WriteResult WriteAsciiChunk(char* data, int size) {
+    const size_t len = static_cast<size_t>(size);
+    size_t off = 0;
+
+    while (off < len && !feof(stream_) && !ferror(stream_))
+      off += fwrite(data + off, 1, len - off, stream_);
+
+    return off == len ? kContinue : kAbort;
+  }
+
+ private:
+  FILE* stream_;
+};
+
+class StreamBaseOutputStream : public AsyncWrap,
```

Comment:
Since this class is specific to heap snapshots, `StreamBaseOutputStream` seems to be a bit too vague..maybe something like `HeapSnapshotSream`? (The factory is named `CreateHeapDumpStream` anyways)

Reply:
done